### PR TITLE
feat: add overlapping rollout training pipeline

### PR DESCRIPTION
**Pull request by inaniloquentee**
## Summary
- Add a bounded local overlap scheduler that runs rollout for the next iteration while training consumes the previous complete rollout batch.
- Add production-facing rollout/training adapters: `RolloutExecutor.generate_candidates(...)` for vLLM rollout and a real PyTorch RL-style optimizer step for DeepSpeed-style local training.
- Connect issue 18 to issue 13's `WeightUpdateManifest` lifecycle through manifest publish/install/release handoff, including timeline metadata and shutdown cleanup.
- Add smoke benchmark modes for synthetic timing, manifest handoff, and realistic WSL2/vLLM rollout plus local training.

## Validation
- `pre-commit run --all-files`
- `py -3.13 -m mypy --ignore-missing-imports rl_engine`
- `py -3.13 -m pytest tests -q`
- `py -3.13 -m pytest rl_engine\tests\test_dispatch.py -v`
- `py -3.13 -m compileall rl_engine tests benchmarks`
- `py -3.13 -m mkdocs build --strict -f mkdocs.yaml`

## Runtime Smoke Results
- `py -3.13 benchmarks\benchmark_overlap_pipeline.py --smoke --mode synthetic --output $env:TEMP\issue18-synthetic.csv --timeline-output $env:TEMP\issue18-synthetic.json`: pass; bounded stale versions `0->1;0->1;1->2`.
- `py -3.13 benchmarks\benchmark_overlap_pipeline.py --smoke --mode manifest-handoff --output $env:TEMP\issue18-manifest.csv --timeline-output $env:TEMP\issue18-manifest.json`: pass; installed versions `[1, 2, 3]`; transport `local-clone`; installed manifests released.
- WSL2/vLLM realistic smoke with local tiny GPT-2 and CPU training: pass through `RolloutExecutor.generate_candidates(...)` plus `TorchRLTrainingWorker`; rollout payload tokens are consumed for training; weight versions `0->1;0->2;1->3`.

## Notes
- This is stacked on issue 13: `issue-13-zero-copy-weight-bridge-rebased`.
- No `task-workspace` issue docs, benchmark CSV artifacts, or timeline JSON artifacts are included.
- Local CI-equivalent commands were run with Python 3.13 because Python 3.10 is not installed locally; GitHub CI will run Python 3.10.

### CODE REVIEW

**Comment by Flink-ddd** [comment]
I believe this PR needs to be validated in user scenarios. If problems arise in the future, we will focus on resolving them, but for now, let's merge them.
